include/aos/common/resourcemonitor.hpp

```diff
@@ -0,0 +1,324 @@
+};
+
+/**
+ * MonitoringData monitoring data.
+ */
+struct MonitoringData {
+    size_t                                        mRAM;
+    size_t                                        mCPU;
+    StaticArray<PartitionInfo, cMaxNumPartitions> mDisk;
+};
+
+/**
+ * InstanceMonitoringData monitoring data for service.
+ */
+struct InstanceMonitoringData {
+    /**
+     * Construct a new Instance Monitoring Data object
+     *
+     */
+    InstanceMonitoringData() = default;
+
+    /**
+     * Construct a new Instance Monitoring Data object
+     *
+     * @param instanceID instance ID
+     * @param monitoringData monitoring data
+     */
+    InstanceMonitoringData(
+        const String& instanceID, const InstanceIdent& instanceIdent, const MonitoringData& monitoringData)
+        : mInstanceID(instanceID)
+        , mInstanceIdent(instanceIdent)
+        , mMonitoringData(monitoringData)
+    {
+    }
+
+    StaticString<cInstanceIDLen> mInstanceID;
+    InstanceIdent                mInstanceIdent;
+    MonitoringData               mMonitoringData;
+};
+
+/**
+ * NodeMonitoringData node monitoring data.
+ */
+struct NodeMonitoringData {
+    StaticString<cNodeIDLen>                              mNodeID;
+    MonitoringData                                        mMonitoringData;
+    time_t                                                mTimestamp;
+    StaticArray<InstanceMonitoringData, cMaxNumInstances> mServiceInstances;
+};
+
+/**
+ * InstanceMonitorParams resource monitor parameters.
+ */
+struct InstanceMonitorParams {
+    InstanceIdent                                 mInstanceIdent;
+    StaticArray<PartitionInfo, cMaxNumPartitions> mPartitions;
+};
+
+/**
+ * NodeMonitorParams resource monitor parameters.
+ */
+struct NodeMonitorParams {
+    StaticString<cNodeIDLen>                      mNodeID;
+    StaticArray<PartitionInfo, cMaxNumPartitions> mPartitions;
+};
+
+/**
+ * Resource usage provider interface.
+ */
+class ResourceUsageProviderItf {
+public:
+    /**
+     * Gatherings system info
+     *
+     * @param systemInfo system info
+     * @return Error
+     */
+    virtual Error GetNodeInfo(NodeInfo& systemInfo) const = 0;
+
+    /**
+     * Gatherings node monitoring data
+     *
+     * @param nodeID node ident
+     * @param monitoringData monitoring data
+     * @return Error
+     */
+    virtual Error GetNodeMonitoringData(const String& nodeID, MonitoringData& monitoringData) = 0;
+
+    /**
+     * Gatherings instance monitoring data
+     *
+     * @param instance instance ident
+     * @param monitoringData monitoring data
+     * @return Error
+     */
+    virtual Error GetInstanceMonitoringData(const String& instanceID, MonitoringData& monitoringData) = 0;
+
+    /**
+     * Gatherings resource config
+     *
+     * @param nodeMonitorParams node monitor params
+     * @return Error
+     */
+    virtual Error GetResourceConfig(NodeMonitorParams& nodeMonitorParams) const = 0;
```

Comment:
It should be in GetNodeInfo shouldn't it?

Reply:
Done, removed this method. Using configuration from GetNodeInfo